### ISSUE TITLE
AddNonSteamGame URI/CLI Support?

### ISSUE THREAD

**Issue by Eoin-ONeill-Yokai** [ValveSoftware/steam-for-linux#7938]
While steam has a relatively flexible URI command system, I feel like this particular part of steam could use an improvement.

I'd like to add the ability for KDE to add a `kdeservices` right click context item for `desktop` files to easily add a non-steam game to steam. Doing so would be relatively easy for the currently logged in steam user if the URI command system allowed for passing desktop file entries to steam, which could then be translated by steam into launcher entries.

There are currently workarounds as specified by [this developer wiki entry](https://developer.valvesoftware.com/wiki/Add_Non-Steam_Game) that are useful for people running their own work around libraries, such as steamsync. However, these work arounds cannot consider the currently logged in user. 

With the news that SteamDeck will be using KDE I figured that this would be a good opportunity to ask whether or not this type of feature could be considered in the future since it could also help with automating the addition of non-steam games installed natively, via lutris or via wine to SteamOS3's big picture mode.

**Comment by deoxal** [comment]
I agree it could use some polish. I have a simple shell script to start a game and it runs fine from the terminal and Lutris, but not on Steam.

The script is this

```
#!/bin/sh

echo A
LD_LIBRARY_PATH=.:${GAME_DATA_PATH}:${LD_LIBRARY_PATH}
export LD_LIBRARY_PATH
./executable
echo B
```
Both A and B get printed but there's no output from Steam in between about what prevented the executable from starting.

**Comment by bertogg** [comment]
The KDE desktop on the Steam Deck comes with an "Add to Steam" action already.

**Comment by Eoin-ONeill-Yokai** [comment]
> The KDE desktop on the Steam Deck comes with an "Add to Steam" action already.

This was posted before the Steam Deck was available. :)
